core/app.go

```diff
@@ -0,0 +1,82 @@
+package core
+
+import (
+	"errors"
+	"sync"
+)
+
+type app struct {
+	m    sync.RWMutex
+	name string
+
+	builds map[string][]Build
+}
+
+// Name is the apps name
+func (a *app) Name() string {
+	if a == nil {
+		return ""
+	}
+
+	return a.name
+}
+
+// Config will apply the config at namespace onto the given conf structure
+// consider this like json.Unmarshal()
+func (a *app) Config(namespace string, conf interface{}) error {
+	return nil
+}
+
+// SendEvent will send the given string on the apps event bus
+func (a *app) SendEvent(event string) {
+
+}
+
+// Listen will provide a channel to select on for a given regular expression
+// returned map is the captured groups and values
+func (a *app) Listen(event string) <-chan map[string]string {
+	return make(chan map[string]string, 1)
+}
+
+func (a *app) NewBuild(group string, config *BuildConfig) (token string, err error) {
+	if a == nil {
+		return "", errors.New("a is nil")
```

Comment:
Where are you running into this (receiver is nil)? Also how is the caller calling a method on a nil value? (Unless you're storing the method itself and calling it with the receiver as the first arg?)